scripts/toolchains/mingw.cmake

```
@@ -1,5 +1,36 @@
 if(NOT _VCPKG_MINGW_TOOLCHAIN)
 set(_VCPKG_MINGW_TOOLCHAIN 1)
+
+#[===[.md:
+# z_vcpkg_message
+Log messages to cmake or a file
+
+```cmake
+z_vcpkg_message(<FATAL_ERROR|...> <message>...)

```

```suggestion
z_vcpkg_mingw_message(<FATAL_ERROR|...> <message>...)

```

If it's going to be mingw specific behavior it seems like that should be named?

---

Hmm, right now it is mingw only, but it can be applied to other toolchains should someone wants to, so I think it's good for now (maybe I'll change it to `z_vcpkg_toolchain_message` or something)

---

This is the spot which benefits from a perspective from outside the mingw context.

Indeed, I regard the focus on mingw as being temporary. 
OTOH the macro can't be moved to `vcpkg.cmake` because sometimes `mingw.cmake` is included directly. It would at least require a fallback defined by `if(NOT COMMAND z_vcpk_message)`...
It might go to another file but probably it also shouldn't be overused.

---

> because sometimes `mingw.cmake` is included directly

Hmm where is it though, I don't remember there's any place that includes `mingw.cmake` directly.

---

You are right, it is not included but chainloaded. And this is probably done via `vcpkg.cmake`:
~~~
$ git grep mingw.cmake
ports/vcpkg-cmake/vcpkg_cmake_configure.cmake:            set(VCPKG_CHAINLOAD_TOOLCHAIN_FILE "${SCRIPTS}/toolchains/mingw.cmake")
scripts/cmake/vcpkg_configure_cmake.cmake:            set(VCPKG_CHAINLOAD_TOOLCHAIN_FILE "${SCRIPTS}/toolchains/mingw.cmake")
scripts/cmake/vcpkg_configure_meson.cmake:            set(VCPKG_CHAINLOAD_TOOLCHAIN_FILE "${SCRIPTS}/toolchains/mingw.cmake")
scripts/cmake/vcpkg_find_fortran.cmake:            set(VCPKG_CHAINLOAD_TOOLCHAIN_FILE "${SCRIPTS}/toolchains/mingw.cmake" PARENT_SCOPE) # Switching to MinGW toolchain for Fortran
~~~

---

Well then we're good with that.

> This is the spot which benefits from a perspective from outside the mingw context.

The way I like to think about it is whether it can be applied to other triplets, and there's one case that I can think of, which is the Linux triplet. Since each distro have its own compiler (and thus its own triplet), it's not exactly cross compiling, but it's also not native compiling either. Because of that, checking the compiler's triplet and see if it's what the user expects (through a variable like `VCPKG_AUTOTOOLS_TRIPLET`) is the way to go, and such we can apply our method here to that.

---

> The way I like to think about it is whether it can be applied to other triplets

Fully agree. 

> `VCPKG_AUTOTOOLS_TRIPLET`

For the record, it already has a name: In another discussion, I was pointed at `VCPKG_MAKE_BUILD_TRIPLET` which is used by `vcpkg_configure_make` (cf. documentation of parameter `BUILD_TRIPLET`).

---

Maybe we can repurpose it as a general Autotools triplet variable then. If that's the case then we need to pass this variable to `get_cmake_vars`, so that the correct compiler is used in `vcpkg_configure_make`
https://github.com/microsoft/vcpkg/blob/030cfaa24de9ea1bbf0a4d9c615ce7312ba77af1/scripts/cmake/vcpkg_internal_get_cmake_vars.cmake#L49-L56

---

Anyway, I'll move the macro to a separate file once I start working on the linux triplets. For now this can serve as a testing ground for the feature.